daemon/graphdriver/windows/windows.go

```diff
@@ -622,33 +630,45 @@ func writeLayerFromTar(r io.Reader, w hcsshim.LayerWriter, root string) (int64,
 
 // importLayer adds a new layer to the tag and graph store based on the given data.
 func (d *Driver) importLayer(id string, layerData io.Reader, parentLayerPaths []string) (size int64, err error) {
-	cmd := reexec.Command(append([]string{"docker-windows-write-layer", d.info.HomeDir, id}, parentLayerPaths...)...)
-	output := bytes.NewBuffer(nil)
-	cmd.Stdin = layerData
-	cmd.Stdout = output
-	cmd.Stderr = output
+	if !noreexec {
+		cmd := reexec.Command(append([]string{"docker-windows-write-layer", d.info.HomeDir, id}, parentLayerPaths...)...)
+		output := bytes.NewBuffer(nil)
+		cmd.Stdin = layerData
+		cmd.Stdout = output
+		cmd.Stderr = output
 
-	if err = cmd.Start(); err != nil {
-		return
-	}
+		if err = cmd.Start(); err != nil {
+			return
+		}
 
-	if err = cmd.Wait(); err != nil {
-		return 0, fmt.Errorf("re-exec error: %v: output: %s", err, output)
-	}
+		if err = cmd.Wait(); err != nil {
+			return 0, fmt.Errorf("re-exec error: %v: output: %s", err, output)
+		}
 
-	return strconv.ParseInt(output.String(), 10, 64)
+		return strconv.ParseInt(output.String(), 10, 64)
+	}
+	return writeLayer(layerData, d.info.HomeDir, id, parentLayerPaths...)
 }
 
-// writeLayer is the re-exec entry point for writing a layer from a tar file
-func writeLayer() {
-	home := os.Args[1]
-	id := os.Args[2]
-	parentLayerPaths := os.Args[3:]
+// writeLayerReexec is the re-exec entry point for writing a layer from a tar file
+func writeLayerReexec() {
+	size, err := writeLayer(nil, os.Args[1], os.Args[2], os.Args[3:]...)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+		os.Exit(1)
+	}
+	fmt.Fprint(os.Stdout, size)
+}
 
-	err := func() error {
+// writeLayer writes a layer from a tar file.
+func writeLayer(layerData io.Reader, home string, id string, parentLayerPaths ...string) (int64, error) {
+	if !noreexec {
+		layerData = os.Stdin
+	}
+	return func() (int64, error) {
 		err := winio.EnableProcessPrivileges([]string{winio.SeBackupPrivilege, winio.SeRestorePrivilege})
```

Comment:
Hmm. `RunWithPrivileges` isn't enough from experimentation as you thought. Hits errors such as the following extracting some files from the tar.

```
 link \\?\C:\control\windowsfilter\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\Files\Windows\servicing\Packages\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat \\?\C:\control\windowsfilter\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\Files\Windows\System32\CatRoot\{F750E6C3-38EE-11D1-85E5-00C04FC295EE}\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat: Access is denied.
time="2016-11-02T16:12:22.349390800-07:00" level=error msg="Error trying v2 registry: failed to register layer: link \\\\?\\C:\\control\\windowsfilter\\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\\Files\\Windows\\servicing\\Packages\\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat \\\\?\\C:\\control\\windowsfilter\\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\\Files\\Windows\\System32\\CatRoot\\{F750E6C3-38EE-11D1-85E5-00C04FC295EE}\\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat: Access is denied."
time="2016-11-02T16:12:22.349390800-07:00" level=error msg="Attempting next endpoint for pull after error: failed to register layer: link \\\\?\\C:\\control\\windowsfilter\\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\\Files\\Windows\\servicing\\Packages\\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat \\\\?\\C:\\control\\windowsfilter\\de4df82af9e94c721e29b37e5753176d991f1e7378939264c5eb2cab0ad585cf\\Files\\Windows\\System32\\CatRoot\\{F750E6C3-38EE-11D1-85E5-00C04FC295EE}\\Package_1007_for_KB3194798~31bf3856ad364e35~amd64~~10.0.1.7.cat: Access is denied."
time="2016-11-02T16:12:27.350605500-07:00" level=info msg="Layer sha256:342d4e407550c52261edd20cd901b5ce438f0b1e940336de3978210612365063 cleaned up"
```

I think this is OK leaving as is for now though as this is a debugging-only feature, and later, I could add a `DisableProcessPrivileges` to go-winio.